File: Mutable/Squadron.m
//
//  Squadron.m
//  Mutable
//
//  Created by Dylan Shackelford on 7/22/15.
//  Copyright (c) 2015 Dylan Shackelford. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "Squadron.h"

@implementation Squadron

+(void)makeTriangleSquadron:(CGPoint)targetPosition container:(UIView *)containerInit placeHolder:(UIButton *)placeHolderButton
{

    //this needs a lot of work, use the double bullet math to make this work
    
    double x = screenWidth + 10;
    double y = arc4random()%(int)screenHeight;
    
    CGPoint aPosition = CGPointMake(x, y);
    
    NSMutableArray* triangleSquadronArray = [[NSMutableArray alloc] init];
    
    Drone* leaderDrone = [[Drone alloc] init];
    
    [leaderDrone initDrone:aPosition Container:containerInit PlaceHolderButton:placeHolderButton];
    
    
    [triangleSquadronArray addObject:leaderDrone];
    
    
    Drone* leftWingDrone = [[Drone alloc] init];
    [leftWingDrone initDrone:CGPointMake(targetPosition.x - 35, targetPosition.y - 35) Container:containerInit PlaceHolderButton:placeHolderButton];
    
    [triangleSquadronArray addObject:leftWingDrone];
    
    
    Drone* rightWingKamikaze = [[Drone alloc] init];
    
    [rightWingKamikaze initDrone:CGPointMake(targetPosition.x + 35, targetPosition.y - 35) Container:containerInit PlaceHolderButton:placeHolderButton];

    
    [triangleSquadronArray addObject:rightWingKamikaze];
}

@end

File: Mutable/GameObjects/MovingObjects/Projectiles/Trackers/Squadron.swift
//
//  Squadron.swift
//  Mutable
//
//  Created by Dylan Shackelford on 1/15/18.
//  Copyright © 2018 Dylan Shackelford. All rights reserved.
//

import Foundation

class Squadron : Drone
{
    func makeTriangleSquadron(targetPosition: CGPoint , containerInit :UIView, placeHolderButton : UIButton)
    {
        let x = 300 //screenwidth + 10
        let y = 100 //arc4random()%screenHeight
        
        let aPosition = CGPoint(x: x, y:y)
        
        let triangleSqaudArray = NSMutableArray()
        
        let leaderDrone:Drone = Drone(drone: aPosition, container: containerInit, placeHolderButton: placeHolderButton)
        triangleSqaudArray.add(leaderDrone)
        
        let leftDrone:Drone = Drone(drone: CGPoint(x:targetPosition.x-35,y:targetPosition.y-35), container: containerInit, placeHolderButton: placeHolderButton)
        triangleSqaudArray.add(leftDrone)
        
        let rightDrone:Drone = Drone(drone: CGPoint(x:targetPosition.x+35,y:targetPosition.y+35), container: containerInit, placeHolderButton: placeHolderButton)
        triangleSqaudArray.add(rightDrone)
    }
}
